lib/core.ps1

```diff
@@ -1300,6 +1300,16 @@ Optimize-SecurityProtocol
 # Load Scoop config
 $configHome = $env:XDG_CONFIG_HOME, "$env:USERPROFILE\.config" | Select-Object -First 1
 $configFile = "$configHome\scoop\config.json"
+# Check if it's the expected install path for scoop: <root>/apps/scoop/current
+$coreRoot = Split-Path $PSScriptRoot
```

Comment:
This in fact should be identical to [`$scoopdir`](https://github.com/ScoopInstaller/Scoop/pull/5369/files#diff-cfd4446b6fd477007943865889b4cdbf84e1c089d960b82cd3b57c6bf019fce3L1336) below, unless the latter is pointed to somewhere else wrong. Maybe we could deprecate `$env:SCOOP` and `confg:rootPath` but I'm afraid that there might be something downstream depends on it, who knows.

Reply:
Better to not touch `$env:SCOOP` I guess. Anyways marked it ready for review.

Reply:
I use `$env:SCOOP` to mark customized path for scoop, and maybe a better way is migrate scoop under `.config` to scoop's own directory, so make scoop portable.

Reply:
Other than the `config.json` file, there is nothing in `~/.config/scoop/`, hence migrating the whole folder does not make sense imho.

If you mean automatical migration, I would say that's not my flavor, since not everyone needs such portability (considering the adoption of the nonportable bucket). The current implementation simply allows existing users to migrate themselves by manually creating/copying the `config.json` when they want. And it's a one-time operation. There is no mandatory migration.

A flag for initialization of portable config could be introduced to the standalone installer. Therefore, new users have a choice to enable it or not.

Reply:
I mean the `config.json` itself, and I could not find a reason that why we put config file in that dir.

Think about single-user PC, all scoop-related files should be in scoop's own dir, and that's portable.

Then multi-users PC, if scoop was installed in personal dir (`USERPROFILE`, as default), the same as single-user's. Or if scoop was installed in shared dir, why use a user specific config file for shared scoop?

Please correct me if I'm mistaken.

Reply:
It was introduced because of the cross-platform concern. It's been stored to `XDG_CONFIG_HOME` therefore it follows [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html). Though Scoop functions completely on Windows only, Scoop has been used for test purposes under *nix platforms, XDG is still a better implementation in this case I believe.

Reply:
That's okay. Now I use scoop under linux by setting env vars and forget this convenient config file.

Reply:
I guess we could migrate it or at least start adding the config to `<root>\config.json` via the installer for new installations.

Reply:
You are syncing scoop via different machine? I know someone is using dot file syncing for scoop's config, and machine-specific settings could be added in config file.

Reply:
No I don't do that, but it could be a feature. It was just an idea. :smile: